Add lines builtin documentation

hdwalters opened the pull request:
Update "0.4.0-alpha" docs only.

Ph0enixKM commented on docs/0.4.0-alpha/advanced_syntax/builtins.md:
````diff
@@ -18,6 +18,30 @@ Transpile to `echo` which prints text to the console, requires a `Text` paramete
+echo len(lines)
+```
+
+N.B. While embedded Bash commands like `$ cat foo.txt $` require a `trust` keyword or `failed` block, Amber does not currently support this for the `lines` builtin.  If the file does not exist at runtime, you will just get a Bash error.
````
Let's make it a warning quote:

```suggestion
> WARNING: While embedded Bash commands like `$ cat foo.txt $` require a `trust` keyword or `failed` block, Amber does not currently support this for the `lines` builtin.  If the file does not exist at runtime, you will just get a Bash error.
```

Ph0enixKM reviewed (changes requested):
@hdwalters one last change: could you add `lines` along with other builtin keywords to the list of keywords?

hdwalters commented:
> @hdwalters one last change: could you add `lines` along with other builtin keywords to the list of keywords?

I compared the current list in that regex against the keyword list I generated from source code, for the Vim plugin. There are quite a few differences; I'll generate a summary table.

hdwalters commented:
> @hdwalters one last change: could you add `lines` along with other builtin keywords to the list of keywords?

Done. Added some keywords, removed others.

Ph0enixKM commented on src/components/Markdown/amber.ts:
```diff
@@ -31,7 +31,7 @@ const amber = (hljs: HLJSApi) => {
     const keywords = {
         scope: 'keyword',
-        match: /\b(ref|to|error|status|if|echo|loop|in|silent|return|fun|else|break|continue|and|or|not|let|sh|main|failed|fail|unsafe|then|pub|import|from|as|nameof|is)\b/
+        match: /\b(and|as|break|cd|continue|echo|else|fail|failed|for|from|fun|if|import|in|is|len|let|loop|main|mv|nameof|not|or|pub|ref|return|silent|status|then|trust|unsafe)\b/
```
@hdwalters it seems that you must have missed `lines` keyword

hdwalters replied:
Oops, good spot.

Ph0enixKM commented on docs/0.4.0-alpha/advanced_syntax/builtins.md:
````diff
@@ -18,6 +18,44 @@ Transpile to `echo` which prints text to the console, requires a `Text` paramete
 ```
 
+## Len
+
+Transpile to `echo "${#TEXT}"` for a `Text` parameter:
+
+```ab
+echo len("Jackdaws love my big sphinx of quartz")
+```
+
+Transpile to `echo "${#ARRAY[@]}"` for an `Array` parameter:
+
+```ab
+echo len(["one", "two", "three", "four", "five"])
+```
````
Let's give the developer more context:

```suggestion

This keyword calculates and returns the length of a given `Text` value, measured in ASCII characters, or the length of an array `[]`. The result is returned as a `Num` type.

For a `Text` parameter, the expression is transpiled internally to the Bash command echo "${#TEXT}". For an `Array` (`[]`) parameter, it is transpiled to echo "${#ARRAY[@]}".

```ab
echo len("Jackdaws love my big sphinx of quartz")
// Returns 37

echo len(["one", "two", "three", "four", "five"])
// Returns 5
```

Ph0enixKM replied:
@hdwalters I thought that we could also add information of what does this built-in do and what type of value does it return

b1ek commented on docs/0.4.0-alpha/advanced_syntax/builtins.md:
````diff
@@ -18,6 +18,46 @@ Transpile to `echo` which prints text to the console, requires a `Text` paramete
+echo len(lines)
+```
+
+> WARNING: While embedded Bash commands like `$ cat foo.txt $` require a `trust` keyword or `failed` block, Amber does not currently support this for the `lines` builtin.  If the file does not exist at runtime, you will just get a Bash error.
````
```suggestion
> WARNING: While embedded Bash commands like `$ cat foo.txt $` require a `trust` keyword or `failed` block, Amber does not currently support this for the `lines` builtin. If the file does not exist at runtime, the program will terminate midway, **potentially losing data stored only in variables**
```

that shit is important

Ph0enixKM commented:
@hdwalters Could we add some general info about the `len`keyword as well?

hdwalters commented:
> @hdwalters Could we add some general info about the `len`keyword as well?

I already imported your suggested text (with some changes as described above). What more did you have in mind?

hdwalters commented:
If you're referring to your comment:

> @hdwalters I thought that we could also add information of what does this built-in do and what type of value does it return

Then surely the latest text covers this: "calculates and returns the length (in ASCII characters) as a `Num` type", and "calculates and returns the length of the array as a `Num` type"?
